sdk/search/azure-search/src/main/java/com/azure/search/SearchServiceUrlParser.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.util.logging.ClientLogger;
+import org.apache.commons.lang3.StringUtils;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * This class is used internally to parse endpoint URLs for use with the underlying AutoRest clients
+ */
+class SearchServiceUrlParser {
+
+    static SearchServiceUrlParts parseServiceUrlParts(String endpoint) {
+        ClientLogger logger = new ClientLogger(SearchServiceUrlParser.class);
+
+        if (StringUtils.isBlank(endpoint)) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Illegal endpoint URL: endpoint cannot be blank"));
+        }
```

Comment:
Would it be fine to remove this and pass a blank/null/empty endpoint into the `URL` constructor if it throws a `MalformedURLException`?